include/node.hpp

```diff
@@ -557,20 +610,45 @@ class merged_node : public node<merged_node<Left, Right, OutId, InId>, meta::con
     }
 
     template<meta::any_simd... Ts>
-        requires meta::vectorizable_v<return_type> && input_port_types::template
-    are_equal<typename std::remove_cvref_t<Ts>::value_type...> && traits::node::can_process_simd<Left>
-            && traits::node::can_process_simd<Right> constexpr stdx::rebind_simd_t<return_type, meta::first_type<meta::typelist<std::remove_cvref_t<Ts>...>>>
-              process_one(Ts... inputs) {
-        return apply_right<InId, traits::node::input_port_types<Right>::size() - InId - 1>(std::tie(inputs...), apply_left<traits::node::input_port_types<Left>::size()>(std::tie(inputs...)));
+        requires traits::node::can_process_simd<Left> && traits::node::can_process_simd<Right>
+    [[gnu::always_inline]] constexpr meta::simdize<return_type, meta::simdize_size_v<std::tuple<Ts...>>>
+    process_one(const Ts &...inputs) {
+        static_assert(traits::node::output_port_types<Left>::size == 1,
+                      "TODO: SIMD for multiple output ports not implemented yet");
+        return apply_right<InId, traits::node::input_port_types<Right>::size() - InId - 1>(
+                std::tie(inputs...), apply_left<traits::node::input_port_types<Left>::size()>(std::tie(inputs...)));
+    }
+
+    [[gnu::always_inline]] constexpr auto
```

Comment:
Will this degrade performance if the forced inline isn't there? This has a high-risk of premature optimisation for the rather simple benchmark cases that we have until now but blow-up once we get to more elaborate benchmark and unit-tests.